UnityProject/Assets/Scripts/Addressables/Types/AddressableReference.cs

```diff
@@ -1,72 +1,46 @@
 ﻿using System;
-using System.Collections;
-using System.Collections.Generic;
 using System.Threading.Tasks;
 using UnityEngine;
 using UnityEngine.AddressableAssets;
-using UnityEngine.ResourceManagement;
-using UnityEngine.ResourceManagement.AsyncOperations;
-using UnityEngine.Serialization;
-using WebSocketSharp;
-
 
 namespace AddressableReferences
 {
 	/// <summary>
 	/// Note about this class, Currently if you want a custom AssetReference Like asset reference texture, might have to make a new class this needs to be explored
 	/// </summary>
 	/// <typeparam name="T"></typeparam>
-	[System.Serializable]
+	[Serializable]
 	public class AddressableReference<T> where T : UnityEngine.Object
 	{
 		public UnLoadSetting SetLoadSetting = UnLoadSetting.KeepLoaded;
-		[FormerlySerializedAs("Path")] public string AssetAddress = "";
+		public string Path = "";
 		public AssetReference AssetReference = null;
 
+		[HideInInspector]
+		public T Asset;
+
 		public bool IsNotValidKey => NotValidKey();
 		public bool IsReadyLoaded => ReadyLoaded();
 
-		private T StoredLoadedReference = null;
-
-
 		#region InternalStuff
 
 		private bool ReadyLoaded()
 		{
+			if ((AssetReference == null) && (!string.IsNullOrWhiteSpace(Path)))
+			{
+				return !(Asset == null);
```

Comment:
Codacy found an issue: [Use the opposite operator ('!=') instead.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)